metadata-integration/java/datahub-client/build.gradle

```diff
@@ -148,7 +148,6 @@ shadowJar {
   relocate 'com.google.errorprone', 'datahub.shaded.com.google.errorprone'
   // Below jars added for kafka emitter only
   relocate 'org.apache.avro', 'datahub.shaded.org.apache.avro'
-  relocate 'org.codehaus.jackson', 'datahub.shaded.org.codehaus.jackson'
```

Comment:
This seems a bit dangerous - have we confirmed that the Datahub client still works after removing jackson?

Reply:
So the reasoning for this is that org.codehaus.jackson has been replaced by com.fasterxml.jackson.core, see here - https://mvnrepository.com/artifact/org.codehaus.jackson/jackson-mapper-asl

There is no longer any reference to org.codehaus.jackson in the codebase (com.fasterxml.jackson.core is used instead).

I do see a couple of tests failed, but that would appear to be as a consequence of other things:
1. Build failed with - "org.neo4j.driver.exceptions.ClientException: The server does not support any of the protocol versions supported by this driver. Ensure that you are using driver and server versions that are compatible with one another."
2. Smoke test (metadata-integration:java:spark-lineage:integrationTest) failed with - "DatahubSparkListener: java.lang.NullPointerException"

I'll try to figure this out, but any help would also be appreciated :)

Reply:
This may actually fix point 2 as well, but I'm unsure ☝️

Reply:
Have increased the version of Neo4jHarness, awaiting review

Reply:
Also increased version of opentelemetry-javaagent.jar to remediate CVE in gson 2.8.6 present in the previously used version

Reply:
Thank you Robert!

Reply:
Running again now :)

Reply:
Looks like the Neo4J upgrade requires Java 11

Reply:
:(

Reply:
I don't think that's correct, both of these suggest that Java 8 should be fine:
1. README/Supported Driver Series - https://github.com/neo4j/neo4j-java-driver/tree/4.4.6
2. Table 1 - https://neo4j.com/docs/java-manual/current/get-started/

That said, there is another error I'll look into now:

/home/runner/work/datahub/datahub/metadata-io/src/test/java/com/linkedin/metadata/graph/neo4j/Neo4jTestServerBuilder.java:5: error: cannot access org.neo4j.graphdb.GraphDatabaseService
import org.neo4j.graphdb.GraphDatabaseService;
                        ^
  bad class file: /home/runner/.gradle/caches/modules-2/files-2.1/org.neo4j/neo4j-graphdb-api/4.4.8/cee96ddf8b21f708ca3456839ca6c67085866aac/neo4j-graphdb-api-4.4.8.jar(org/neo4j/graphdb/GraphDatabaseService.class)
    class file has wrong version 55.0, should be 52.0
    Please remove or make sure it appears in the correct subdirectory of the classpath.

Reply:
Apologies, I am confusing the Neo4j driver and Neo4j itself, this error is indeed because using Neo4j/harness v 4.x is compiled with Java 11 ☝️.

I'm struggling to find any clear cut confirmation but some things that come up from a Google suggest that It may be that changing harness to the latest 3.5.x version will work so have changed that. Awaiting another review (thanks for being patient both!)

Reply:
A question for either of you if you know the answer - com.google.code.gson-gson-2.8.6.jar and org.pac4j-pac4j-core-3.8.3.jar get into the datahub-frontend image?

From the dockerfile it would seem that one of these two commands somehow gets/builds these jars, but I can't figure out where the version is defined (the versions of those jars in the docker image have CVEs too):
./gradlew :datahub-web-react:build -x test -x yarnTest -x yarnLint \
./gradlew :datahub-frontend:dist -PenableEmber=${ENABLE_EMBER} -PuseSystemNode=${USE_SYSTEM_NODE} -x test -x yarnTest -x yarnLint \

Reply:
@jjoyce0510 @RyanHolstien I'm figuring this out by trawling Maven for the dependencies ☝️ 

Any idea when this PR might get another review/re-run of the tests

Reply:
@jjoyce0510 @RyanHolstien So unfortunately the latest pipeline failure is a result of versions of pac4j-oidc 4+ not being compatible with Java 8. This is a bit annoying as there are some CVEs that therefore can't be resolved until a version 3.x of this package gets fixes (if it ever will), but I've changed the version to the latest 3.x version (which at least fixes some CVEs) and built the code locally.

Another review/re-run would be great when someone has the time